src/hera/workflow.py

```diff
@@ -127,8 +127,8 @@ def __init__(
         active_deadline_seconds: Optional[int] = None,
         metrics: Optional[Union[Metric, List[Metric], Metrics]] = None,
     ):
-        self.name = validate_name(name)
-        dag_name = self.name if dag_name is None else dag_name
+        self.name = validate_name(name, generate_name=generate_name)
+        dag_name = self.name.rstrip("-.") if dag_name is None else dag_name
```

Comment:
What do you think about making this part of the validation process as well? :) It might be useful to tell the user that `-.` or `.-` is not accepted the same way `_` is not accepted, which is reported by the validator

Reply:
Added explicit validation and custom error message for trailing `.` and `-` in `validators.py`.

Just to be clear, the addition of `rstrip("-.")` in `workflow.py` was to prevent unexpected validation error when an otherwise valid `name` would be invalid for use in `dag_name`.